DashAISoftware/DashAI
# Pydantic-like DataLoader Schemas

# Summary


New Pydantic Schemas for Dataloaders.

## Type of change



- Back end new feature.
- Refactoring.

## Changes


- New _string_field_, for raw strings.
- Refactor old _string_field_ into _enum_field_.
- Refactor models and _test_schemas_ to use _enum_field_.
- Add schemas for CSVDataLoader and JSONDataLoader.
- Add schemas for splits and more options inside BaseDataloader file.
- Add method to replace the **$defs** field inside the respective field.

## How to Test


Run `tests\back\schemas\test_schemas.py` to check the correct refactorization.
Run `tests\back\dataloader\test_dataloader_schema.py` to check the generation of the schemas.
